checkov/terraform/checks/graph_checks/aws/APIGatewayEndpointsUsesCertificateForAuthentication.yaml

```diff
@@ -0,0 +1,11 @@
+metadata:
+  id: "CKV2_AWS_51"
+  name: "Ensure AWS API Gateway endpoints uses client certificate authentication"
+  category: "GENERAL_SECURITY"
+definition:
+  cond_type: "attribute"
+  resource_types:
+  - "aws_api_gateway_stage"
+  - "aws_apigatewayv2_stage"
```

Comment:
for API Gateway V2 you can only set `client_certificate_id`, if the actual API protocol type is websocket. So you would need to check for the connection to an `aws_apigatewayv2_api` and its `protocol_type`

Reply:
thanks for pointing it out, I missed it in terraform documentation. I will update the yaml check accordingly.

Reply:
@gruebel Updated check according to the comment and updated pass fail test cases.